### PR TITLE
Check UUID in read adapter equals() and hashCode() methods

### PR DESCRIPTION
**Pull request by droazen**

### CODE REVIEW

**Comment by droazen** [comment]
For @davidadamsphd  to review.

**Comment by davidadamsphd** [comment]
Looks like you have some test problems. Let me know when they're fixed.

**Comment by droazen** [comment]
To fix the failing tests, I had to add an `equalsIgnoreUUID()` method to the `GATKRead` interface. Tests should pass now -- please re-review at your leisure.

File: src/main/java/org/broadinstitute/hellbender/utils/read/GoogleGenomicsReadToGATKReadAdapter.java [hunk 1]
```diff
@@ -635,20 +635,26 @@ public Read convertToGoogleGenomicsRead() {
...
     @Override
-    public boolean equals( Object o ) {
-        if ( this == o ) return true;
-        if ( o == null || getClass() != o.getClass() ) return false;
+    public boolean equalsIgnoreUUID( final Object other ) {
+        if ( this == other ) return true;
+        if ( other == null || getClass() != other.getClass() ) return false;
 
-        GoogleGenomicsReadToGATKReadAdapter that = (GoogleGenomicsReadToGATKReadAdapter) o;
+        GoogleGenomicsReadToGATKReadAdapter that = (GoogleGenomicsReadToGATKReadAdapter)other;
```
Comment (davidadamsphd) [inline src/main/java/org/broadinstitute/hellbender/utils/read/GoogleGenomicsReadToGATKReadAdapter.java]: nit: space before other.

**Comment by davidadamsphd** [comment]
LGTM.

**Pull request merged**